block/src/Pos/Block/Logic/Creation.hs

```diff
@@ -447,7 +447,11 @@ createMainBody bodyLimit sId payload =
                 psks' <- takeSome psks
                 usPayload' <- includeUSPayload
                 return (psks', usPayload')
-        let dlgPay' = leftToPanic "createMainBlockPure: " $ mkDlgPayload psks'
+        let dlgPay' = UnsafeDlgPayload psks'
+        -- TBD: is it necessary to check here?
+        -- What if it fails? What will be the behaviour of cardano-sl at
+        -- large?
```

Comment:
If it fails, it's a programmer error. So there should be `error` call in that case. That's how it works in `develop`.
This check is done just in case, since performance was not an issue so far.
I'd remove this check here now, because performance matters now.